sopn_publish_date/__init__.py

```diff
@@ -19,23 +23,50 @@ def __init__(self):
             "gla": self.greater_london_assembly,
             "pcc": self.police_and_crime_commissioner,
             "mayor": self.mayor,
-            "parl": self.uk_parliament,
         }
         self.calendar = UnitedKingdomBankHolidays()
 
-    def for_id(self, election_id: str) -> date:
+    def for_id(self, election_id: str, country: str = None) -> date:
         """
-        Calculate the publish date for an election given in `uk-election-ids <https://elections.democracyclub.org.uk/reference_definition/>`_ format, or raise an exception if that election id is ambiguous (could correspond to elections in multiple countries with different electoral legislation)
+        Calculate the publish date for an election given in `uk-election-ids <https://elections.democracyclub.org.uk/reference_definition/>`_ format and an optional country if necessary (for example, local or parliamentary elections), or raise an exception if that election id is ambiguous (could correspond to elections in multiple countries with different electoral legislation)
 
         :param election_id: a string representing an election id in uk-election-ids format
+        :param country: an optional string representing the country where the election will be held
         :return: a datetime representing the expected publish date
         """
+
         election_type, poll_date = type_and_poll_date(election_id)
 
+        countries = {
+            "ENG": Country.ENGLAND,
+            "WLS": Country.WALES,
+            "SCT": Country.SCOTLAND,
+            "NIR": Country.NORTHERN_IRELAND,
+        }
+
+        def valid_election_type(el_type):
+            return el_type in self.election_id_lookup or el_type in ["local", "parl"]
+
+        def requires_country(el_type):
+            return el_type not in self.election_id_lookup
+
+        if not valid_election_type(election_type):
+            raise NoSuchElectionTypeError(election_type)
```

Comment:
Do you reckon `NoSuchElectionTypeError` needs to be a different exception from `InvalidElectionIdError`? They both basically boil down to "this string isn't a senisble identifier", although one is a slightly more specific variation of that.

Bit of a tangent, but one of the things I'd like to do is define a single picture-check validator for IDs in the `uk-election-ids` package ( https://github.com/DemocracyClub/uk-election-ids/issues/3 ) which would be useful here, and various other places.

Reply:
This is purely a taste thing - at some point I should introduce a parent exception class, but I tend to err on the side of specific exceptions.